src/main/java/com/internet/shop/dao/impl/ProductDaoImpl.java

```diff
@@ -0,0 +1,45 @@
+package com.internet.shop.dao.impl;
+
+import com.internet.shop.dao.ProductDao;
+import com.internet.shop.db.Storage;
+import com.internet.shop.lib.Dao;
+import com.internet.shop.model.Product;
+import java.util.List;
+import java.util.NoSuchElementException;
+import java.util.Optional;
+
+@Dao
+public class ProductDaoImpl implements ProductDao {
+    @Override
+    public Product create(Product product) {
+        Storage.addProduct(product);
+        return product;
+    }
+
+    @Override
+    public Optional<Product> get(Long id) {
+        return Storage.products.stream()
+                .filter(x -> x.getId().equals(id))
+                .findFirst();
+    }
+
+    @Override
+    public List<Product> getAll() {
+        return Storage.products;
+    }
+
+    @Override
+    public Product update(Product product) {
+        for (int i = 0; i < Storage.products.size(); i++) {
+            if (Storage.products.get(i).getId().equals(product.getId())) {
+                return Storage.products.set(i, product);
+            }
+        }
+        throw new NoSuchElementException("Such product was not found");
```

Comment:
let's pass product id to exception message

Reply:
added